### ISSUE TITLE
Correctness of the implementation

### ISSUE THREAD

**Issue by quark0** [sayakpaul/EvoNorms-in-TensorFlow-2#1]
Thanks for trying out these layers! I'm one of the authors of the paper.

If I'm reading your code correctly, `v`, `beta`, and `gamma` are of the shape (1, C, 1, 1) ([link](https://github.com/sayakpaul/EvoNorms-in-TensorFlow-2/blob/c31974dfd37337c32a12393544e1adfdbbc5c27b/layer_utils/EvoNorm2dS0.py#L22)) but the input is of the shape (N, H, W, C) ([link](https://github.com/sayakpaul/EvoNorms-in-TensorFlow-2/blob/c31974dfd37337c32a12393544e1adfdbbc5c27b/layer_utils/EvoNorm2dS0.py#L7)). The mismatch (channel axis=1 for the former but 3 for the latter) would make the implementations for both layers incorrect (e.g., in this case one will be normalizing along the height instead of the channel dim for S0). Can you double check?

**Comment by sayakpaul** [comment]
You are right, thanks for pointing it out. Let me try it out and get me back to you. @quark0

**Comment by sayakpaul** [comment]
Hi @quark0. I just updated the set of experiments. Happy to address any more feedback you might have.

**Comment by quark0** [comment]
Thanks for fixing it. I didn't check everything, but another issue I can spot at the moment is that the moving var for B0 should take the shape of (1, 1, 1, C) instead of (1, C, 1, 1) ([link](https://github.com/sayakpaul/EvoNorms-in-TensorFlow-2/blob/6552709fb03e7a6fe454d291671a17badcec249e/layer_utils/EvoNorm2dB0.py#L40)). I'm a bit surprised that [this line](https://github.com/sayakpaul/EvoNorms-in-TensorFlow-2/blob/6552709fb03e7a6fe454d291671a17badcec249e/layer_utils/EvoNorm2dB0.py#L46) didn't trigger any shape incompatibility error though—it seems you're attempting to take the max of two tensors with shapes (1, C, 1, 1) and (N, H, W, C) if my understanding is correct.

**Comment by sayakpaul** [comment]
Thanks for your inputs, @quark0. I just pushed [the updated implementation](https://github.com/sayakpaul/EvoNorms-in-TensorFlow-2/blob/master/layer_utils/EvoNorm2dB0.py) and notebooks. Also, earlier I was taking standard deviation instead of variance in this particular case. I corrected the same.

**Comment by quark0** [comment]
Thanks for updating the code. I have two more thoughts:
1. I believe `self.eps` still needs to be added to `var` before taking `tf.sqrt` ([line](https://github.com/sayakpaul/EvoNorms-in-TensorFlow-2/blob/b5f6b45eb90578cfaa3269fdab37ba351c0bf05a/layer_utils/EvoNorm2dB0.py#L45)), similar to what we do for BatchNorms. This is likely going to result in NaNs otherwise.
2. Regarding (`momentum`, `eps`), it seems their values are (0.9, 1e-5) for EvoNorms but (0.99, 1e-3) for BatchNorms ([TF2.0 default](https://www.tensorflow.org/api_docs/python/tf/keras/layers/BatchNormalization)). This would introduce nontrivial differences in terms of both the quality and stability. I'd suggest we use (0.99, 1e-3) in both cases.

nit: These [four lines](https://github.com/sayakpaul/EvoNorms-in-TensorFlow-2/blob/b5f6b45eb90578cfaa3269fdab37ba351c0bf05a/layer_utils/EvoNorm2dB0.py#L36-L39) could be simplified as:
```
_, var = tf.nn.moments(x, [0, 1, 2], keepdims=True)
```

**Comment by sayakpaul** [comment]
That makes sense and I updated the code accordingly. Pushed a notebook with the updated results as well. However, for S0 I stayed with `eps` to be `1e-5`.

**Comment by quark0** [comment]
Thanks a lot for spending time to improve the implementation.

**Comment by sayakpaul** [comment]
Of course, thank you so much for helping me to do that. I will always remember this :)
